### PR TITLE
Bugfix/426 add category filter

### LINKED ISSUE #426: Add category filter parameter to location group and time series group getAll endpoints
This will allow filtering to only groups for specific categories (like "Agency Aliases")

### PR DESCRIPTION
**Pull request by rma-rripken**
Closes #426

### CODE REVIEW

File: cwms-data-api/src/main/webapp/regexp.html [hunk 1]
```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE html>
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/webapp/regexp.html:1]: Great addition.

**Review by MikeNeilson — APPROVED** [review]

File: cwms-data-api/src/main/webapp/regexp.html (hunk 1 above)
Comment (rma-rripken) [reply cwms-data-api/src/main/webapp/regexp.html:1]: If users need help figuring out how to write something tricky in Oracle syntax we can add to these examples.

Comment (rma-rripken) [reply cwms-data-api/src/main/webapp/regexp.html:1]: Actually, maybe we should say that - Request help from HEC if you would like help crafting a regular expression or have suggested examples to add to this doc.  I'm gonna merge this now and let you figure out how to add to or fix that part

Comment (MikeNeilson) [reply cwms-data-api/src/main/webapp/regexp.html:1]: I would word that as "As for help from HEC or open a new discussion at <link to github discussion>" but yes that's a good idea.

**Pull request merged**